Marlin/src/module/planner.cpp

```diff
@@ -2056,7 +2056,7 @@ bool Planner::_populate_block(block_t * const block, bool split_move,
     #if ENABLED(LIN_ADVANCE)
 
       #if ENABLED(JUNCTION_DEVIATION)
-        #define MAX_E_JERK (max_e_jerk_factor * max_acceleration_mm_per_s2[_EINDEX])
```

Comment:
This cannot be changed, because `_EINDEX` relies on the `extruder` parameter passed to `_populate_block`. We cannot use `active_extruder`, which could very well have been changed before this function was called.

Reply:
The problem is that for the calculation to be correct the max_acceleration_mm_per_s2 term needs to be SQRT(max_acceleration_mm_per_sz[_EINDEX]), but I assume that it is not ideal to be calling SQRT here? Is this the case?

If it is undesirable to call SQRT in this context would it be acceptable to create an array of pre-computed SQRT values for max_acceleration_mm_per_s2 (updated if the underlying values is changed), one for each extruder? I'm not that familiar with the Marlin space/time trade offs but this would add (in the worst case) a second static array of floats with one entry per extruder. If you think this is OK I'll go ahead and modify the PR to do things that way.

Reply:
In my opinion the best thing is to use an array for max_e_jerk [EXTRUDERS], so calculate the value for each extruder with its corresponding acceleration and use the macro #define MAX_E_JERK max_e_jerk [tools.active_extruder]

Reply:
@gloomyandy — A pre-computed array would be do the trick. The array will only needed when `DISTINCT_E_FACTORS` is set. Otherwise a single computed value will do.

Reply:
I've just pushed a change that should handle multiple extruders correctly. I did not special case the single extruder case though. Is the compiler smart enough to optimize a single element array accessed via constant value indices into just a single value?

Reply:
When `DISTINCT_E_FACTORS` is disabled, no array is required, and `max_acceleration_mm_per_s2` only has a single E entry. So I've patched the code to properly handle this case.